app/controllers/signs_controller.rb

```diff
@@ -1,15 +1,18 @@
 # frozen_string_literal: true
 
-class SignsController < ApplicationController
-  require 'open-uri'
+require 'open-uri'
 
+class SignsController < ApplicationController
   before_action :find_vocab_sheet, :set_search_query, :footer_content
 
   def search
     search_query = process_search_query
     @page_number = permitted_params[:p].present? ? permitted_params[:p].to_i : 1
     @results_total, @signs = Sign.paginate(search_query, @page_number)
     @query = search_query
+    @pagination_html = SignPaginationService.new(current_page_number: @page_number,
```

Comment:
Maybe this should be accessed by a helper for the HTML view, instead of a variable on the controller.